components/freertos/ringbuf.c

```diff
@@ -409,6 +409,16 @@ static void returnItemToRingbufBytebuf(ringbuf_t *rb, void *item) {
     //Free the read memory.
     rb->free_ptr=rb->read_ptr;
 }
+/*
+    find out if semaphore returned from xQueueSelectFromSet belong to this xRingbuffer
+*/
+BaseType_t xRingbufferCanRead(RingbufHandle_t ringbuf, QueueHandle_t sem)
+{
+    ringbuf_t *rb=(ringbuf_t *)ringbuf;
+    configASSERT(rb);
+    return rb->items_buffered_sem == sem;
```

Comment:
``return (rb->items_buffered_sem == (SemaphoreHandle_t) member) ? pdTRUE : pdFALSE;`` corresponding to the ``QueueSetMemberHandle_t member`` parameter change suggested above.